packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4742,3 +4743,31 @@ export class Gitxnas extends Gtxnas {
 		super.execute(stack);
 	}
 }
+
+/**
+ * Takes the last value from stack and if base64encoded, decodes it and push it back to the stack,
+ * othervise throws an error
+ * https://github.com/algorand/go-algorand/pull/3220
+ */
+export class Base64Decode extends Op {
+	readonly line: number;
+	/**
+	 * Asserts 0 arguments are passed.
+	 * @param args Expected arguments: [] // none
+	 * @param line line number in TEAL file
+	 */
+	constructor(args: string[], line: number) {
+		super();
+		this.line = line;
+		assertLen(args.length, 0, line);
+	}
+
+	execute(stack: TEALStack): void {
+		this.assertMinStackLen(stack, 1, this.line);
+		const last = this.assertBytes(stack.pop(), this.line);
+		assertBase64(convertToString(last), this.line);
+		const enc = new TextDecoder("utf-8");
```

Comment:
Change logic to support `URLEncoding` and `StdEncoding`.
I'm also don't know exactly method to do it 😭